azurelinuxagent/common/osutil/default.py

```diff
@@ -58,6 +61,25 @@ def __init__(self):
     def get_agent_conf_file_path(self):
         return self.agent_conf_file_path
 
+    def get_instance_id(self):
+        '''
+        Azure records a UUID as the instance ID
+        First check /sys/class/dmi/id/product_uuid.
+        If that is missing, then extracts from dmidecode
+        If nothing works (for old VMs), return the empty string
+        '''
+        if os.path.isfile(PRODUCT_ID_FILE):
+            return fileutil.read_file(PRODUCT_ID_FILE).strip()
+
+        rc, s = shellutil.run_get_output(DMIDECODE_CMD)
+        if rc != 0:
```

Comment:
Probably worth checking for 'Not Settable' as well, in case this is not supported; in that case the return code will still be 0.